pytorch_lightning/trainer/trainer.py

```diff
@@ -838,7 +838,25 @@ def call_setup_hook(self, model):
         self.setup(stage_name)
         model.setup(stage_name)
 
+    def _reset_result_and_set_hook_fx_name(self, hook_name):
+        model_ref = self.get_model()
+        if model_ref is not None:
+            # used to track current hook name called
+            model_ref._results = Result()
+            model_ref._current_hook_fx_name = hook_name
+
+    def _cache_logged_metrics(self):
+        model_ref = self.get_model()
+        if model_ref is not None:
+            # capture logging for this hook
+            self.logger_connector.cache_logged_metrics()
+
     def call_hook(self, hook_name, *args, **kwargs):
+        # temporary. Don't modify evaluation behaviour
+        if self.logger_connector._current_stage == "train":
```

Comment:
shall we later consider using Enum for stages to avoid typos, if yes, lest create an issue for it... :]

Reply:
Yes, definitely. I will do this.